reconcile/test/test_openshift_rolebindings.py

```diff
@@ -0,0 +1,348 @@
+            body={
+                "apiVersion": "rbac.authorization.k8s.io/v1",
+                "kind": "RoleBinding",
+                "metadata": {
+                    "name": "test-role5-test-namespace5-test-serviceaccount",
+                },
+                "roleRef": {
+                    "kind": "Role",
+                    "name": "test-role5",
+                },
+                "subjects": [
+                    {
+                        "kind": "ServiceAccount",
+                        "name": "test-serviceaccount",
+                        "namespace": "test-namespace5",
+                    },
+                ],
+            },
+        ),
+        resource_name="test-role5-test-namespace5-test-serviceaccount",
+        privileged=False,
+    )
+
+
+def test_should_add_desired_state_cluster_not_present() -> None:
+    # ri does not have the cluster
+    ri = ResourceInventory()
+    ri.initialize_resource_type(
+        "test-cluster", "test-namespace", "RoleBinding.rbac.authorization.k8s.io"
+    )
+    test_role = get_app_interface_test_roles()
+    role_binding_specs = RoleBindingSpec.create_rb_specs_from_role(test_role[0], None)
+    assert not should_add_desired_state(
+        ri, role_binding_specs[0].get_oc_resources()[0], role_binding_specs[0]
+    )
+
+
+def test_should_add_desired_state_cluster_present_and_resource_not_present() -> None:
+    ri = ResourceInventory()
+    ri.initialize_resource_type(
+        "test-cluster5", "test-namespace5", "RoleBinding.rbac.authorization.k8s.io"
+    )
+    test_role = get_app_interface_test_roles()
+    role_binding_specs = RoleBindingSpec.create_rb_specs_from_role(test_role[0], None)
+    assert should_add_desired_state(
+        ri, role_binding_specs[0].get_oc_resources()[0], role_binding_specs[0]
+    )
+
+
+def test_should_add_desired_state_cluster_present_and_resource_present() -> None:
+    ri = ResourceInventory()
+    ri.initialize_resource_type(
+        "test-cluster5", "test-namespace5", "RoleBinding.rbac.authorization.k8s.io"
+    )
+    ri.add_desired(
+        cluster="test-cluster5",
+        namespace="test-namespace5",
+        resource_type="RoleBinding.rbac.authorization.k8s.io",
+        name="test-role5-test-org-user",
+        value=OR(
+            integration="openshift-rolebindings",
+            integration_version="0.3.0",
+            error_details="test-role5-test-org-user",
+            body={
+                "kind": "RoleBinding",
+                "apiVersion": "rbac.authorization.k8s.io/v1",
+                "metadata": {
+                    "name": "test-role5-test-org-user",
+                },
+            },
+        ),
+    )
+    test_role = get_app_interface_test_roles()
+    role_binding_specs = RoleBindingSpec.create_rb_specs_from_role(test_role[0], None)
+    assert not should_add_desired_state(
+        ri, role_binding_specs[0].get_oc_resources()[0], role_binding_specs[0]
+    )
```

Comment:
this means we still rely on `ri` to be pre populated to get desired state, what's the business case that we need to skip certain clusters? these can be added to test cases of `fetch_desired_state` instead of testing util function `should_add_desired_state`